cli/src/main.rs

```diff
@@ -181,7 +182,14 @@ fn download_program_fast(shared_options: &SharedOptions, path: &str) -> Result<(
 
         let mm = session.target.memory_map.clone();
 
-        download_file(&mut session, std::path::Path::new(&path), Format::Elf, &mm)?;
+        let progress = FlashProgress::new(|_| {});
+        download_file(
+            &mut session,
+            std::path::Path::new(&path),
+            Format::Elf,
+            &mm,
+            &progress,
```

Comment:
It would be nice if the caller could choose if they want to supply a progress callback or no, so I would prefere something like an option here, or two separate methods.